python/triton/compiler.py

```diff
@@ -699,6 +699,28 @@ def visit_Call(self, node):
     def visit_Constant(self, node):
         return triton.language.constexpr(node.value)
 
+    def visit_BoolOp(self, node: ast.BoolOp):
+        assert len(node.values) == 2
+        lhs = self.visit(node.values[0])
+        rhs = self.visit(node.values[1])
+        if isinstance(lhs, triton.language.constexpr):
```

Comment:
Just curious, if there is logic like `a or b and c`, will we handle the operator precedence issue properly?

Reply:
It seems that the AST module takes care of the operator precedence.

e.g.

```python
import ast
tree = ast.parse("var = a or b and c")
pprint(ast.dump(tree))
```
it will get 

```python
Assign('var', value=BoolOp(op=Or(), values=['a', BoolOp(op=And(), values=['b', 'c'])])
```
AST also takes good care of braces, 

```python
tree = ast.parse("var = (a or b) and c")
```
it will get
```python
Assign('var', value=BoolOp(op=And(), values=[BoolOp(op=Or(), values=['a','b']), 'c'])
```